src/components/Header/Header.tsx

```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+import clsx from 'clsx';
+
+import {HeaderProps} from './Header.types';
+import './Header.scss';
+import {Separator, Typography} from '~/components';
+import {ArrowLeft} from '~/icons';
+
+export const Header: React.FC<HeaderProps> = ({
+    title,
+    search = null,
+    mainActions,
+    status,
+    contentType,
+    toolbarRight,
+    toolbarLeft,
+    backButton = null,
+    breadcrumb,
+    className,
+    ...props
+}: HeaderProps) => {
+    const hasInformationArea = Boolean(breadcrumb || contentType || status);
+    const hasToolbar = Boolean(toolbarLeft || toolbarRight);
+
+    return (
+        <header className={clsx('moonstone-header', className)} {...props}>
+            {/* Main area */}
+            <div className={clsx('moonstone-header_main', 'flexRow', 'alignCenter', 'flexFluid')}>
+                { backButton && (
+                    <backButton.type
+                        {...backButton.props}
+                        className={clsx(backButton.props.className, 'moonstone-header_back')}
+                        variant="outlined"
+                        icon={<ArrowLeft/>}
+                    />
+                )}
+
+                <Typography isNowrap component="h1" variant="title" className={clsx('flexFluid', 'moonstone-header_title')}>{title}</Typography>
+
+                { search && (
+                    <div className={clsx('moonstone-header_search')}>
+                        { search }
+                    </div>
+                )}
+
+                { mainActions && (
+                    <div className={clsx('moonstone-header_mainActions', 'moonstone-header_actions')}>
+                        { mainActions }
+                    </div>
+                )}
+            </div>
+
+            { hasInformationArea && (
+                <div className={clsx('flexRow_between', 'alignCenter', 'moonstone-header_information')}>
+                    <div className={clsx('flexRow_nowrap', 'alignCenter', 'flexFluid', 'moonstone-header_informationLeft')}>
+                        { breadcrumb }
```

Comment:
Isn't this very specific to a type of header editing content ? Don't we have use cases where a more generic "information area" could be required ?

Reply:
I guess it depends if we want to restrict to this type of information area...

Reply:
From what I saw, we don't have other example for an information area needs. The idea is to facilitate the implementation to manage spacing, separator from the component itself. Plus if we want to modify where the elements are positioned it will not impact the implementation.